examples/petsc/bpsswarm.c

```diff
@@ -0,0 +1,439 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and other CEED contributors.
+// All Rights Reserved. See the top-level LICENSE and NOTICE files for details.
+//
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This file is part of CEED:  http://github.com/ceed
+
+//                        libCEED + PETSc Example: CEED BPs
+//
+// This example demonstrates a simple usage of libCEED with PETSc to solve the CEED BP benchmark problems, see http://ceed.exascaleproject.org/bps, on
+// a particle swarm.
+//
+// The code uses higher level communication protocols in DMPlex and DMSwarm.
+//
+// Build with:
+//
+//     make bpsswarm [PETSC_DIR=</path/to/petsc>] [CEED_DIR=</path/to/libceed>]
+//
+// Sample runs:
+//
+//     bpssphere -problem bp1 -degree 3
+//     bpssphere -problem bp2 -degree 3
+//     bpssphere -problem bp3 -degree 3
+//
+//TESTARGS -ceed {ceed_resource} -test -problem bp3 -dm_plex_dim 3 -dm_plex_box_faces 10,10,10 -dm_plex_simplex 0 -swarm uniform -points_per_cell 500
```

Comment:
Can maybe decrease the problem size a bit so it's not expensive. (Coarser mesh and 64 points per element seems like enough.)

Reply:
Okay, I had to tweak error tolerances, but I got the problem sizes much smaller. I also added other BPs to make sure they also work fine.